File: WittyShare/WittyShare/Models/Article.cs
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WittyShare.Models
{
    [BsonIgnoreExtraElements]
    public class Article
    {
        public Article(int articleId, Person creator, string title, string content, string patentStatus, IEnumerable<Comment> comments, IEnumerable<string> categories, int amazing, int good, int poor, IEnumerable<Person> favouritedBy, string image)
        {
            ArticleId = articleId;
            Creator = creator;
            Title = title;
            Content = content;
            PatentStatus = patentStatus;
            Comments = comments;
            Categories = categories;
            Amazing = amazing;
            Good = good;
            Poor = poor;
            FavouritedBy = favouritedBy;
            Image = image;
        }
        
        public int ArticleId { get; set; }
        public Person Creator { get; set; }
        public string Title { get; set; }
        public string Content { get; set; }
        public string PatentStatus { get; set; }
        public IEnumerable<Comment> Comments { get; set; }
        public IEnumerable<string> Categories { get; set; }
        public int Amazing { get; set; }
        public int Good { get; set; }
        public int Poor { get; set; }
        public IEnumerable<Person> FavouritedBy { get; set; }
      public string Image { get; set; }
    }
}

File: WittyShare/WittyShare/ClientApp/app/components/models/article.ts
import { Person } from './person';
import { Comment } from './comment';
export class Article {
    articleId: number;
    creator: Person;
    title: string;
    content: string;
    patentStatus: string;
    amazing: number;
    good: number;
    poor: number;
    comments: any=[];
    categories: any=[];
    favouritedBy: any=[];

    image: string;
    constructor(data: any) {
        if (data) {
            data = data || {};
            this.articleId = data.articleId;
            this.creator = new Person(data.creator);
            this.title = data.title;
            this.content = data.content;
            this.patentStatus = data.patentStatus;
            this.amazing = data.amazing;
            this.good = data.good;
            this.poor = data.poor;
            this.image = data.image;
            data.comments.forEach((c: any) => {
                this.comments.push(new Comment(c));
            });
            data.categories.forEach((o: any) => {
                this.categories.push(o);
            });
            data.favouritedBy.forEach((p: any) => {this.favouritedBy.push(new Person(p));
            });
        }
    }
}


